src/sentry/identity/vsts/provider.py

```diff
@@ -254,7 +252,7 @@ def exchange_token(self, request: Request, pipeline, code):
                     "client_id": self.client_id,
                     "client_secret": self.client_secret,
                     "code": code,
-                    "redirect_uri": absolute_uri(pipeline.redirect_url()),
+                    "redirect_uri": pipeline.config.get("redirect_url"),
```

Comment:
this is always set in the vsts integration (same above): https://github.com/getsentry/sentry/blob/bc5cfde2c7333cc1874d8c0afbe4df21f8164d88/src/sentry/integrations/vsts/integration.py#L452